flaml/integrations/oai/completion.py

```diff
@@ -187,34 +188,40 @@ def eval(cls, config: dict, prune=True, eval_only=False):
         prune_hp = cls._prune_hp
         metric = cls._metric
         config_n = config[prune_hp]
-        max_tokens = config["max_tokens"]
+        max_tokens = config.get("max_tokens", 16)  # default value is 16
         region_key = cls._get_region_key(config)
         prompt = cls._prompts[config["prompt"]]
         stop = cls._stops and cls._stops[config["stop"]]
         if prune and target_n_tokens:
             max_valid_n = cls._get_max_valid_n(region_key, max_tokens)
-            min_invalid_n = cls._get_min_invalid_n(region_key, max_tokens)
-            if min_invalid_n is not None and config_n >= min_invalid_n:
-                if config_n > max_valid_n:
+            if cls.avg_input_tokens:
+                # max_tokens bounds the maximum tokens
+                # so using it we can calculate a valid n according to the avg # input tokens
+                max_valid_n = max(
+                    max_valid_n,
+                    int((target_n_tokens - cls.avg_input_tokens) // max_tokens),
+                )
```

Comment:
One question: this does not need to be applied to min_invalid_n?

Reply:
Right. max_tokens is only the upper limit.